src/init.cpp

```diff
@@ -252,10 +245,7 @@ void Shutdown()
 #endif
     UnregisterAllValidationInterfaces();
 #ifdef ENABLE_WALLET
-    for (CWalletRef pwallet : vpwallets) {
-        delete pwallet;
-    }
-    vpwallets.clear();
+    DeleteWallets();
```

Comment:
Yes, `DeleteWallets()` is a bad name. `DeallocWallets()` is good, or perhaps `DetachWallets()`